test/functional/collection_tests.js

```diff
@@ -1250,10 +1250,15 @@ describe('Collection', function() {
     });
   });
 
-  it('should allow an empty replacement document for findOneAndReplace', function() {
+  it('should allow an empty replacement document for findOneAndReplace', function(done) {
     const configuration = this.configuration;
     const client = configuration.newClient({}, { w: 1 });
 
+    let finish = err => {
```

Comment:
this really makes me wonder if we can come up with a better approach to all of this. What if we had a plug-in for mocha that allowed us to register deferred actions before done is called, like:
```
it(‘should do whatever’, function(done) {
  const client = new MongoClient(...);
  done.defer(() => client.close());
  ... rest of test...
});
```

Reply:
100% agree.